pdns/ws-auth.cc

```diff
@@ -1709,6 +1709,31 @@ static void patchZone(HttpRequest* req, HttpResponse* resp) {
             if (rr.qtype.getCode() == QType::SOA && rr.qname==zonename) {
               soa_edit_done = increaseSOARecord(rr, soa_edit_api_kind, soa_edit_kind);
             }
+
+            // Check if the DNSNames that should be hostnames, are hostnames
+            if (rr.qtype.getCode() == QType::NS || rr.qtype.getCode() == QType::MX || rr.qtype.getCode() == QType::SRV) {
+              DNSName toCheck;
+              if (rr.qtype.getCode() == QType::SRV) {
+                vector<string> parts;
+                stringtok(parts, rr.getZoneRepresentation());
+                if (parts.size() == 4) toCheck = DNSName(parts[3]);
+              } else if (rr.qtype.getCode() == QType::MX) {
+                vector<string> parts;
+                stringtok(parts, rr.getZoneRepresentation());
+                if (parts.size() == 2) toCheck = DNSName(parts[1]);
+              } else {
+                toCheck = DNSName(rr.content);
+              }
+
+              if (toCheck.empty()) {
+                throw ApiException("RRset "+qname.toString()+" IN "+qtype.getName() + " unable to extract hostname from content.");
```

Comment:
Perhaps "Bad Request" is a better error here?

Reply:
AFAIK the rest of the API will return 400 only when the parsing of the JSON failed. I would prefer to stick to the "Unprocessable entity" error, but can change it of course!

Reply:
Please stick to 422 :)